network/src/noise/handshake.rs

```diff
@@ -205,14 +213,19 @@ impl NoiseUpgrader {
             HandshakeAuthMode::ServerOnly => None,
         };
 
+        // send [self_peer_id | expected_public_key] as prologue
+        let mut prologue = self.self_peer_id.to_vec();
+        prologue.extend_from_slice(remote_public_key.as_slice());
+        socket.write_all(&prologue).await?;
```

Comment:
i wonder if it makes sense to combine prologue and first_message into a single buffer and send op? we use TCP_NODELAY so the two writes will probably get split...

Reply:
good point!